examples/Extension/ProgressLogger/CatLogger.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace PhpBench\Examples\Extension\ProgressLogger;
+
+use PhpBench\Model\Benchmark;
+use PhpBench\Model\Iteration;
+use PhpBench\Model\Subject;
+use PhpBench\Model\Suite;
+use PhpBench\Model\Variant;
+use PhpBench\Progress\LoggerInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CatLogger implements LoggerInterface
+{
+    /**
+     * @var OutputInterface|null
+     */
+    private $output;
+
+    public function setOutput(OutputInterface $output)
+    {
+        $this->output = $output;
+    }
+
+    public function variantStart(Variant $variant)
+    {
+    }
+
+    public function startSuite(Suite $suite)
+    {
+    }
+
+    public function retryStart(int $rejectionCount)
+    {
+    }
+
+    public function iterationStart(Iteration $iteration)
+    {
+        $this->output->write('🐈');
+    }
+
+    public function iterationEnd(Iteration $iteration)
+    {
+    }
+
+    public function subjectStart(Subject $subject)
+    {
+    }
+
+    public function variantEnd(Variant $variant)
+    {
+    }
+
+
+    public function subjectEnd(Subject $subject)
+    {
+    }
+
+    public function benchmarkStart(Benchmark $benchmark)
+    {
+    }
+
+    public function benchmarkEnd(Benchmark $benchmark)
+    {
+    }
+
+    public function endSuite(Suite $suite)
+    {
+    }
+}
```

Comment:
the API here is pretty awful in many ways:

- Some of these argument-objects are mutable.
- The `OutputInterface` is set via. setter injection could make `OututInteface` a service, or even write directly to STDOUT.